minpy/dispatch/rule.py

```diff
@@ -253,3 +268,37 @@ def _get_arg_rule_key(self, args, kwargs):
         arg_key = [self._get_type_signiture(x) for x in args]
         kwarg_key = sorted(kwargs.keys())
         return '-'.join(arg_key) + '+' + '-'.join(kwarg_key)
+
+    @classmethod
```

Comment:
In fact they cannot create different rules. The rule is associated with class instead of objects. Ending with this solution is a compromise of all existing features like `with` statement of a policy object. I've emphasized in the doc.